src/main/java/com/bobocode/hoverla/bring/context/ApplicationContextImpl.java

```diff
@@ -1,60 +1,152 @@
+import java.util.stream.Collectors;
+import java.util.Objects;
+
 
 /**
- * Example of ApplicationContext implementation, can be removed or reimplemented when actual development starts
+ * Implementation of ApplicationContext
  */
+@Slf4j
 public class ApplicationContextImpl implements ApplicationContext {
+    private Table<String, Class<?>, BeanDefinition> context = HashBasedTable.create();
+    private final String BEAN_TYPE_MUST_BE_NOT_NULL_MESSAGE = "The argument [beanType] must be not null";
+    private final String NO_SUCH_BEAN_EXCEPTION_MESSAGE = "Bean with provided name/type [%s] not found in the context";
+
+    /**
+     *  Scanners scan application to define beans configs and map them into {@link BeanDefinition} lists.
+     *  Received list of bean definitions validates by validator to avoid duplicates and other problems (see {@link BeanValidator}).
+     *  Beans definitions list maps to container structure. It's Table<String, Class<?>, BeanDefinition>, where
+     *      - row key is a name of bean
+     *      - column key is a type of bean
+     *      - mapped value is bean definition
+     *  Then context passes to {@link BeanInitializer} for beans instances initialization
+     *
+     * @param scanners      list of scanners for
+     * @param validator     bean definition validator
+     * @param initializer   bean initializer
+     */
     public ApplicationContextImpl(List<BeanScanner> scanners,
                                   BeanValidator validator,
                                   BeanInitializer initializer) {
-        /* 1) Scanners get all beans
-           2) Merge List of bean definitions into one List
-           3) Send beans list to validator to check duplicates
-           4) Convert bean list to map
-           5) Send beans map to initializer to instantiate beans and inject dependencies
-         */
+
+        List<BeanDefinition> beanDefinitionList = scanningApplication(scanners);
+        validate(validator, beanDefinitionList);
+        contextCreation(beanDefinitionList);
+        beansInstantiation(initializer);
+
+        log.info("Application context initialized");
+    }
+
+    private List<BeanDefinition> scanningApplication(List<BeanScanner> scanners) {
+        List<BeanDefinition> beanDefinitionList = new ArrayList<>();
+        log.info("Scanning application for finding bean definitions");
+        for (BeanScanner beanScanner : scanners) {
+            log.debug("Running bean scanner {}", beanScanner.getClass().getSimpleName());
+            beanDefinitionList.addAll(beanScanner.scan());
+        }
+        return beanDefinitionList;
+    }
+
+    private void validate(BeanValidator validator, List<BeanDefinition> beanDefinitionList) {
+        log.info("Beans definitions validating");
+        validator.validate(beanDefinitionList);
+    }
+
+    private void contextCreation(List<BeanDefinition> beanDefinitionList) {
+        log.debug("Context table generation");
```

Comment:
I would rephrase this log message to `Store scanned bean definitions in application context`